src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-edit-form/directions-institution-hierarchies-edit-form.component.ts

```diff
@@ -0,0 +1,174 @@
+    private dialogRef: MatDialogRef<DirectionsInstitutionHierarchiesEditFormComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private store: Store) {
+    this.store.dispatch(new GetDirections());
+    this.lastInsHierarchy = this.getLastInsHierarchy();
+    this.buildForm();
+  }
+
+  ngOnInit(): void {
+  }
+
+  ngAfterViewInit(): void {
+    setTimeout(() => {
+      if (this.editForm) {
+        this.lastInputFocus();
+      }
+    }, 0);
+  }
+
+  buildForm() {
+    let directions = [...this.lastInsHierarchy.directions];
+    const formGroupFields = this.getFormControlsFields();
+    formGroupFields[this.userDirectionsLabel] = new FormControl(directions);
+    this.EditDirectionFormGroup = new FormGroup(formGroupFields);
+    this.directionsControl = this.EditDirectionFormGroup.get(this.userDirectionsLabel) as FormControl;
+  }
+
+  getDirections() {
+    let lastInsHierarchy = this.data.element.insHierarchies[this.data.element.insHierarchies.length - 1];
+    return lastInsHierarchy.directions;
+  }
+
+  getLastInsHierarchy() {
+    return this.data.element.insHierarchies[this.data.element.insHierarchies.length - 1];
+  }
+
+  getFormControlsFields() {
+    const formGroupFields = {};
+    formGroupFields[this.ministryLabel] = new FormControl({value: this.data.element.insHierarchies[0].institution.title, disabled: true});
+    this.fields.push(this.ministryLabel);
+    let field, title;
+    for (let i = 0; i < this.data.columns.length; ++i) {
+      field = this.data.columns[i];
+      title = this.data.element.insHierarchies[i]?.title;
+      if (title) {
+        formGroupFields[field] = new FormControl(title);
+      }
+      else {
+        formGroupFields[field] = new FormControl({value: null, disabled: true});
+      }
+      this.fields.push(field);
+    }
+    return formGroupFields;
+  }
+
+  onCancel() {
+    this.closeDialog();
+  }
+
+  onSubmit() {
+    for (let i = 0; i < this.data.columns.length; ++i) {
+      const fieldName = this.data.columns[i];
+      const field = this.EditDirectionFormGroup.controls[fieldName];
+      if (field.value != this.data.element.name[i]) {
+        let editedInsHierarchy = this.data.element.insHierarchies[i];
+        editedInsHierarchy.title = field.value;
+        this.editedInsHierarchies.push(editedInsHierarchy);
+      }
+    }
+
+    if (!this.compareTwoArrays(this.directionsControl.value, this.lastInsHierarchy.directions)) {//(this.directionsControl.value != this.lastInsHierarchy.directions) {
+      let editedInsHierarchy = this.editedInsHierarchies.find(ins => ins.id === this.lastInsHierarchy.id);
+      if (editedInsHierarchy) {
+        editedInsHierarchy.directions = this.directionsControl.value;
+      }
+      else {
+        editedInsHierarchy = this.lastInsHierarchy;
+        editedInsHierarchy.directions = this.directionsControl.value;
+        this.editedInsHierarchies.push(editedInsHierarchy);
+      }
+    }
+
+    forkJoin(
```

Comment:
It is better to pass updated data as a result and update institution in `DirectionsInstitutionHierarchiesListComponent`